aviatrix/utils.go

```diff
@@ -110,8 +100,8 @@ func extractTags(d *schema.ResourceData, cloudType int) (map[string]string, erro
 	if !ok {
 		return nil, nil
 	}
-	if !intInSlice(cloudType, []int{goaviatrix.AWS, goaviatrix.AWSGOV, goaviatrix.AZURE}) {
-		return nil, fmt.Errorf("adding tags is only supported for AWS, AWSGOV and AZURE, cloud_type must be 1, 256 or 8")
+	if !intInSlice(cloudType, []int{goaviatrix.AWS, goaviatrix.AWSGOV, goaviatrix.GCP, goaviatrix.AZURE}) {
+		return nil, fmt.Errorf("adding tags is only supported for AWS, AWSGOV, GCP and AZURE, cloud_type must be 1, 256, 4 or 8")
 	}
 
 	tagsMap := tags.(map[string]interface{})
```

Comment:
GCP needs its own regex for tag validation. Something like ```regexp.Compile(`^[\p{Ll}\p{Lo}\p{N}_-]*$`)```. I copied the regex from Google's page, but I removed the check for tag length. Checking for tag length for other cloud types requires much more logic which I think is better left in the backend.